fl4health/utils/metrics.py

```diff
@@ -228,3 +228,29 @@ def compute(self) -> Dict[str, Scalar]:
     def clear(self) -> None:
         self.metric_values_history = [[] for _ in range(len(self.metrics))]
         self.counts = []
+
+
+class MetricMeterManager:
+    """
+    Class to manage one or metric meters.
+    """
+
+    def __init__(self, key_to_meter_map: Dict[str, MetricMeter]):
+        self.key_to_meter_map = key_to_meter_map
+
+    def update(self, preds: Dict[str, torch.Tensor], target: torch.Tensor) -> None:
+        for map_key, pred_key in zip(sorted(self.key_to_meter_map.keys()), sorted(preds.keys())):
```

Comment:
So we may have a bit of an issue that we've seen before with zip. That is, if the keys are not of equal lengths, zip will still function fine and maybe even pass the assert below if `pred.keys()` is a subset of `key_to_meter_map.keys()`, which will cause some unexpected behavior. So we should also assert that they key sets are the same length

Reply:
I was assuming that these would be equal lengths and contain the same keys for this but I will add that edge case

Reply:
Yeah it's just an extra sanity check. We've been burned by the zip on lists of unequal length thing before so I'm paranoid 😂

Reply:
Yeah, I was planning on including more thorough checks in the follow up PR where we have the user define the metermanager because then we have no guarantees about what the keys they pass are and if they map back to what is returned by the predict method. Now we at least have some control because we are defining the manager in the constructor. But I can add some initial checks now but it will probably change a bit next PR

Reply:
Updated in [9959c89](https://github.com/VectorInstitute/FL4Health/pull/62/commits/9959c896eba8b13846f86a313be23b34619dc1cb)